core/src/style/textStyle.cpp

```diff
@@ -95,6 +96,20 @@ Label::Options TextStyle::optionsFromTextParams(const Parameters& _params) const
     return options;
 }
 
+const std::string& TextStyle::applyTextSource(const Parameters& _parameters, const Properties& _props) const {
+
+    if (_parameters.textSource.text.empty()) {
+        // Default: use 'name' property
+        return _props.getString("name");
+    }
```

Comment:
Oh, this test must also check for !_parameters.textSource.isFunction, because the function could have returned an empty string